### PR TITLE
add metrics for function duration for listing gce instances and metrics for inconsistent mig count when gce bulk instance listing is enabled

### PR DESCRIPTION
**Pull request by damikag**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
This PR emits function duration metrics for listing all mig instances and listing instances using ```igm.ListManagedInstances()```, when bulk listing gce instances is enabled.

And also emits inconsistent mig count for the number of migs which have a difference in instance count based on actionsSummary and result from bulk listing instances.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?
NONE

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: cluster-autoscaler/cloudprovider/gce/mig_info_provider.go [hunk 1]
```diff
@@ -245,6 +249,13 @@ func (c *cachingMigInfoProvider) isMigCreatingOrDeletingInstances(mig Mig) bool
 
 // updateMigInstancesCache updates the mig instances for each mig
 func (c *cachingMigInfoProvider) updateMigInstancesCache(migToInstances map[GceRef][]GceInstance) error {
+	listingStartTime := time.Now()
+	defer func() { metrics.UpdateDuration(metrics.BulkListMigInstances, time.Since(listingStartTime)) }()
+	inconsistentInstancesMigsCount := 0
+	defer func() {
+		klog.V(1).Infof("Inconsistent instances migs count: %v", inconsistentInstancesMigsCount)
```
Comment (bskiba) [inline cluster-autoscaler/cloudprovider/gce/mig_info_provider.go:256]: Warning?

Comment (damikag) [reply cluster-autoscaler/cloudprovider/gce/mig_info_provider.go:256]: Done

File: cluster-autoscaler/cloudprovider/gce/mig_info_provider.go [hunk 2]
```diff
@@ -196,6 +197,9 @@ func (c *cachingMigInfoProvider) listInstancesInAllZonesWithMigs() ([]GceInstanc
 	var allInstances []GceInstance
 	errors := make([]error, len(zones))
 	zoneInstances := make([][]GceInstance, len(zones))
+	listingStartTime := time.Now()
+	defer func() { metrics.UpdateDuration(metrics.BulkListAllGceInstances, time.Since(listingStartTime)) }()
```
Comment (x13n) [inline cluster-autoscaler/cloudprovider/gce/mig_info_provider.go:201]: nit: `defer metrics.UpdateDurationFromStart(metrics.BulkListAllGceInstances, time.Now())` would work just fine here.

File: cluster-autoscaler/cloudprovider/gce/mig_info_provider.go [hunk 3]
```diff
@@ -245,6 +249,13 @@ func (c *cachingMigInfoProvider) isMigCreatingOrDeletingInstances(mig Mig) bool
 
 // updateMigInstancesCache updates the mig instances for each mig
 func (c *cachingMigInfoProvider) updateMigInstancesCache(migToInstances map[GceRef][]GceInstance) error {
+	listingStartTime := time.Now()
+	defer func() { metrics.UpdateDuration(metrics.BulkListMigInstances, time.Since(listingStartTime)) }()
```
Comment (x13n) [inline cluster-autoscaler/cloudprovider/gce/mig_info_provider.go:253]: nit: ditto, you can use `UpdateDurationFromStart` and there's no need to wrap into `func()`.

**Review by x13n — CHANGES_REQUESTED** [review]
Looks good, just minor comments.

File: cluster-autoscaler/cloudprovider/gce/mig_info_provider.go (hunk 2 above)
Comment (damikag) [reply cluster-autoscaler/cloudprovider/gce/mig_info_provider.go:201]: Updated

File: cluster-autoscaler/cloudprovider/gce/mig_info_provider.go (hunk 3 above)
Comment (damikag) [reply cluster-autoscaler/cloudprovider/gce/mig_info_provider.go:253]: Updated

**Pull request merged**